torchgeo/datamodules/flair2.py

```diff
@@ -0,0 +1,74 @@
+"""This module contains the FLAIR2DataModule class for loading the FLAIR2 dataset.
+
+The FLAIR dataset is released under open license 2.0:
+- https://www.etalab.gouv.fr/wp-content/uploads/2018/11/open-licence.pdf
+- https://ignf.github.io/FLAIR/#FLAIR2
+
+Code for loading dataset licensed under the MIT License.
+"""
+
+from typing import Any
+
+import kornia.augmentation as K
+import torch
+from torch.utils.data import random_split
+
+from ..datasets import FLAIR2
+from ..samplers.utils import _to_tuple
+from ..transforms import AugmentationSequential
+from .geo import NonGeoDataModule
+
+
+class FLAIR2DataModule(NonGeoDataModule):
+    """LightningDataModule implementation for the FLAIR2 dataset.
+
+    Uses the train/test splits from the dataset.
+    """
+
+    def __init__(
+        self,
+        batch_size: int = 64,
+        patch_size: tuple[int, int] | int = 64,
+        val_split_pct: float = 0.2,
+        num_workers: int = 0,
+        augs: AugmentationSequential | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize a new FLAIR2DataModule instance.
+
+        Args:
+            batch_size: Size of each mini-batch.
+            patch_size: Size of each patch, either ``size`` or ``(height, width)``.
+                Should be a multiple of 32 for most segmentation architectures.
+            val_split_pct: Percentage of the dataset to use as a validation set.
+            num_workers: Number of workers for parallel data loading.
+            augs: Optional augmentations to apply to the dataset.
+            **kwargs: Additional keyword arguments passed to
+                :class:`~torchgeo.datasets.Potsdam2D`.
+        """
+        super().__init__(FLAIR2, batch_size, num_workers, **kwargs)
+
+        self.patch_size = _to_tuple(patch_size)
```

Comment:
I am by no means an expert, but have been reading some papers in the domain the last year. I suppose, the increased feature dimensions provided by the hyperspectral bands of sentinel-2 have shown to increase model performance.

Furthermore, in the datapaper [\[p. 6\]](https://arxiv.org/pdf/2305.14467) of FLAIR2 the authors argue: 

> The interest of the extended spatial information provided by the Sentinel-2 super-patches is particularly visible in the last two rows of Figure 4. Indeed, the location on a beach or on a lake is difficult to determine from the aerial image alone, and could easily be confused with the sea for example in the last row. 

The model leveraging the additional information outperform a standard baseline model:

| Model        | mIoU  | Building | Pervious Surface | Impervious Surface | Bare Soil | Water  | Coniferous | Deciduous | Brushwood | Vineyard | Herbaceous Vegetation | Agricultural Land | Plowed Land |
|--------------|-------|----------|------------------|---------------------|-----------|--------|------------|-----------|-----------|----------|------------------------|-------------------|-------------|
| U-Net        | 0.547 | 0.8009   | 0.4727           | 0.6988              | 0.3076    | 0.7985 | 0.5758     | 0.7014    | 0.2392    | 0.6012   | 0.4653                | 0.5449            | **0.3583**  |
| U-T&T        | 0.5594| 0.8285   | 0.4980           | 0.7204              | 0.2982    | **0.8009** | 0.6041 | 0.7189    | **0.2541** | 0.6580   | 0.4684                | 0.5478            | 0.3157      |
| U-T&T best   | **0.5758** | **0.8368** | **0.4995**     | **0.7446**          | **0.3959** | 0.7952 | **0.6339** | **0.7239** | 0.2485 | **0.6678** | **0.4750**        | **0.5513**        | 0.3381      |

Reply:
Thanks for pointing that out, I should have looked at the paper myself... In that case, if we want to support a resize, it would be more involved, if we wont to resize the sentinel image as well but keeping the resolution difference. Or still just provide the Sentinel 2 scene as context.

Reply:
I think that's a good idea. Could either be included here, or in a separate PR.